autorest/azure/rp.go

```diff
@@ -0,0 +1,190 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// DoRetryWithRegistration tries to register the resource provider in case it is unregistered.
+// It also handles request retries
+func DoRetryWithRegistration(client autorest.Client) autorest.SendDecorator {
+	return func(s autorest.Sender) autorest.Sender {
+		return autorest.SenderFunc(func(r *http.Request) (resp *http.Response, err error) {
+			shouldRetry := true
+			rr := autorest.NewRetriableRequest(r)
+			for shouldRetry {
+				err = rr.Prepare()
+				if err != nil {
+					return resp, err
+				}
+
+				resp, err = autorest.SendWithSender(s, rr.Request(),
+					autorest.DoRetryForStatusCodes(client.RetryAttempts, client.RetryDuration, autorest.StatusCodesForRetry...),
+				)
+				if err != nil {
+					return resp, err
+				}
+				shouldRetry = false
+
+				if resp.StatusCode == http.StatusConflict {
```

Comment:
After removing the shouldRetry you can change this to exit early on the non-conflicting status.
```go
if resp.StatusCode != http.StatusConflict {
    return resp, err
}
var re RequestError
...
```

Reply:
done :3